types/transaction.go

```diff
@@ -35,6 +35,43 @@ type Transaction struct {
 	TxMetadata string `json:"tx_metadata"`
 }
 
+func (t *Transaction) Equal(b *Transaction) bool {
```

Comment:
Not a big fan of this if cascade 🙃
Another option is to create a RawTransaction struct. Something like this:
```go
type Transaction struct {
        RawTransaction `gorm:"embedded"`
	// Id is the unique identifier for this transaction
	Id string `json:"id"`
	// ParentId is the parent transaction id
	ParentId string `json:"parent_id"`
}

type RawTransaction struct {
        BasicBlockData `gorm:"embedded"`
	// Level is the nested level of the transaction
	Level uint16 `json:"level"`
	// TxTimestamp is the timestamp of the transaction
	TxTimestamp time.Time `json:"tx_timestamp"`
	// TxCid is the transaction hash
	TxCid string `json:"tx_cid" gorm:"index:idx_transactions_tx_hash"`
	// TxFrom is the sender address
	TxFrom string `json:"tx_from" gorm:"index:idx_transactions_tx_from"`
	// TxTo is the receiver address
	TxTo string `json:"tx_to" gorm:"index:idx_transactions_tx_to"`
	// Amount is the amount of the tx in attoFil
	Amount *big.Int `json:"amount" gorm:"type:numeric"`
	// GasUsed is the total gas used amount in attoFil
	GasUsed uint64 `json:"gas_used"`
	// Status
	Status string `json:"status"`
	// TxType is the message type
	TxType string `json:"tx_type" gorm:"index:idx_tx_type"`
	// TxMetadata is the message metadata
	TxMetadata string `json:"tx_metadata"`
}
```
This way the comparison could be performed at `RawTransaction` level:
 * aTx.RawTransaction == bTx.RawTransaction

Reply:
Not sure how this behave at db level :/

Reply:
I would not change that since the indexer is using that struct. But, when comparing why not keep the original `Transaction` struct and the compare func could first set `Id = ""` and `ParentId= ""` so if those fields change for whatever reason, they won't be compared anyways